extensions/amp-list/0.1/amp-list.js

```diff
@@ -179,21 +194,39 @@ export class AmpList extends AMP.BaseElement {
       if (maxLen < items.length) {
         items = items.slice(0, maxLen);
       }
-      return this.renderItems_(items);
+      return this.scheduleRender_(items);
     }, error => {
       throw user().createError('Error fetching amp-list', error);
     });
   }
 
   /**
+   * Schedules a fetch result to be rendered in the near future.
    * @param {!Array} items
    * @return {!Promise}
    * @private
    */
-  renderItems_(items) {
-    return this.templates_.findAndRenderTemplateArray(this.element, items)
+  scheduleRender_(items) {
+    let resolver;
+    const promise = new Promise(resolve => {
+      resolver = resolve;
+    });
+    this.renderQueue_.push({resolver, items});
+    this.renderPass_.schedule();
+    return promise;
+  }
+
+  /**
+   * Dequeues a fetch result and renders it immediately.
+   * @private
+   */
+  doRenderPass_() {
+    dev().assert(this.renderQueue_.length > 0, 'Nothing queued to render.');
+    const {resolver, items} = this.renderQueue_.shift();
+    this.templates_.findAndRenderTemplateArray(this.element, items)
```

Comment:
There is a small change of behaviour here. Previously if any of `findAndRenderTemplateArray`, `updateBindings_` or `rendered_` methods failed, the reject promise would make it all the way to the caller of layoutCallback, now it will just hang. 

I don't know if any of those can fail in legit cases, but resource manager handles layoutCallback rejection in a special way (removed loading icon, etc..)

Might be worth passing `rejector` along as well.

Reply:
Good point, thanks.